### ISSUE TITLE
_recorder - matching http headers as well as URL

### ISSUE THREAD

**Issue by stevelittlefish** [getsentry/responses#639]
For my project, I need to mock responses from an API where some of the parameters are included in a HTTP header, so for example I would make a request to `http://api.com/projects` with a http header set to `user-id: joebloggs`

Currently I am recording the responses by doing:

```python
@_recorder.record(filepath="out.yaml")
def test_something():
    ...
```

and then I'm loading the yaml file like this:

```python
@responses.activate
def test_something():
    responses.add_from_file("out.yaml")
    ...
```

Is it possible to set up the registry to match on this HTTP header, and also save this header in the .yaml files that it generates?

**Comment by stevelittlefish** [comment]
Yes I think so (although I haven't actually done it like that using this library as I have just been using the recorded files up until now).

I'm pretty sure I would want to do something like:
```
    responses.get(
        url="http://loginsystem.com/projects",
        match=[matchers.header_matcher({"Impersonate-User": "Mr. Bean"})],
    )
```

**Comment by beliaev-maksim** [comment]
Gotcha


Unfortunately, I cannot commit to any timeline, since it is pretty hard to accommodate all the matchers for the recorder. Need to think about some nice API that could be extended and that provides all the functionality for recording into file
